tsl/src/compression/compression.c

```diff
@@ -2062,27 +2040,17 @@ decompress_batches_for_insert(ChunkInsertState *cis, Chunk *chunk, TupleTableSlo
 }
 
 #if PG14_GE
-/*
- * Helper method which returns true if given column_name
- * is configured as SEGMENT BY column in a compressed hypertable
- */
-static bool
-is_segmentby_col(List *ht_compression_info, char *column_name)
+static void
+add_filter_column_strategy(char *column_name, StrategyNumber strategy, Const *value,
+						   List **filter_columns, List **filter_ops, List **filter_columns_value,
+						   List **is_null_check)
 {
-	ListCell *lc;
-	foreach (lc, ht_compression_info)
-	{
-		FormData_hypertable_compression *fd = lfirst(lc);
-		if (namestrcmp(&fd->attname, column_name) == 0)
-		{
-			if (fd->segmentby_column_index > 0)
-				return true;
-			break;
-		}
-	}
-	return false;
+	*filter_columns = lappend(*filter_columns, column_name);
```

Comment:
This is at the chunk level so we don't need to keep name and can resolve it to AttributeNumber